### PR TITLE
Add node to track Hubitat hub mode

### PR DESCRIPTION
**Pull request by bkeifer**
I registered the mode callback with a deviceId of 0 to fit it in with the way you've been doing the callback lookups.  It's a bit of a hack, but should be safe from any ID collisions.

### CODE REVIEW

File: nodes/mode.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        return parseFloat(value);
+      case "BOOL":
+        return value == "true";
+      default:
+        console.warn("Unable to cast to dataType. Open an issue to report back the following output:");
+        console.warn(dataType);
+        console.warn(value);
+        return value;
+    }
+  }
```
Comment (fblackburn1) [inline nodes/mode.js]: You can remove it

File: nodes/config.js [hunk 2]
```diff
@@ -132,7 +149,11 @@ module.exports = function(RED) {
       return;
     }
 
-    const callback = callbacks[req.body.content["deviceId"]];
+    if(req.body.content["deviceId"] != null) {
+      var callback = callbacks[req.body.content["deviceId"]];
+    } else if (req.body.content["deviceId"] == null && req.body.content["name"] == "mode") {
```
Comment (fblackburn1) [inline nodes/config.js]: Avoid redundancy condition
```suggestion
    } else if (req.body.content["name"] == "mode") {
```

File: nodes/mode.js (hunk 1 above)
Comment (bkeifer) [reply nodes/mode.js]: Agreed.  Will remove in the next commit.

**Comment by bkeifer** [comment]
I wrestled with how to handle the extra fields we get from an event vs. what is returned from a GET to the Maker API modes/ endpoint.

In the end, I left the extra fields in because they may be of use to someone in the future.  I'm not sure descriptionText will be all that helpful, but I think displayName will help people who need to distinguish between multiple homes or hubs.

Given that, what do you think about updating the documentation to something like the following:

Screen Shot 2020-02-19 at 2 10 56 PM

**Comment by fblackburn1** [comment]
I'm ok with that :+1:
In this case, can you rewrite the payload of the event output to only have these four fields?

**Pull request merged**